stdlib/array.ncl

```diff
@@ -254,7 +269,7 @@
         "%m
       = fun f n => %generate% n f,
 
-    sort : forall a. (a -> a -> [| Less, Equal, Greater |]) -> Array a -> Array a
+    sort : forall a. (a -> a -> ComparisonResult) -> Array a -> Array a
```

Comment:
Is it not statically typechecked? The (static) type checker currently reads this as an opaque type, why doesn't it complain?

Reply:
This is why the CI is failing currently, I missed this one. Indeed we need a contract application.

Reply:
I don't understand your answer, I'm afraid :slightly_smiling_face: